### DISCUSSION TITLE
Making str and List[LiteralString] play nicely together with the `index` method

### DISCUSSION THREAD

**Discussion by ciphergoth** [microsoft/pyright#6821] [Q&A]
```python3
listlit = "foo bar baz".split()

def listlitix(x : str):
    return listlit.index(x) 

print(listlitix("bar"))
```
gives me an error on the parameter of `index`: 
```
Argument of type "str" cannot be assigned to parameter "__value" of type "LiteralString" in function "index"
  "str" is incompatible with "LiteralString"Pylance[reportGeneralTypeIssues](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportGeneralTypeIssues)
```

What should I be doing to make this typecheck correctly?

**Comment by erictraut** [comment]
See https://github.com/python/typeshed/issues/10887 for information about this problem.

**Comment by michaelbutleridt** [comment]
Does anyone have a user side solution for this? I have a simple piece of code that pyright says is an error; of course fixing from the pyright side would be great, but I'd be happy to add an annotation or cast to fix this if possible.

Here's a small snippet example: 

Code sample in [pyright playground](https://pyright-play.net/?code=JYWwDg9gTgLgBAKwM4QHYCh0BtioKZJwC8cA2gERp7kA0c5MA7hLfTABZR7UC6mAxHAAmAV3CEuMEVFSEAhnCQwo6AG5ysxRClQA6UeIAUOfEgCUAuGDlQ5IQsEJLgWTQqUrrt%2B1vIBjCBAQOVQhJCIAbwBfcl0AM2hgmEN1LAt0AKCQoV85RickEEU8UIBaTODQuFLSxnY5GDxVPChyDMDKnJIK7N0kMBxk8noLHtDdOTAwEqFDchqvOyRyUY7eyenQw0X7CzAoXGSxoQsgA)

```python
import json

lines = ["one", "two", "three"]

# dumps returns a str
val = json.dumps(lines)

# params is still a str
params = "commands={}".format(val)

command = "aws ssm send-command --whatever"
command = command.split(" ")
command.append("--params")
command.append(params)  # pyright complains here because str is not compat. w/ LiteralString?
print(command)

```

It was not clear to me from this discussion and the linked TypeShed issue how to fix this in my code. Thanks.

**Reply by erictraut** [reply]
I think this is something that will need to be addressed in either the typeshed stubs or the type system. Pyright's current behavior is conformant with the type system and the type information it is provided.

**Reply by michaelbutleridt** [reply]
OK, thanks. For now I found a brute force workaround: 

```
# type: ignore
```

I can add that at the end of the complaining line.
